### ISSUE TITLE
Coverage tests fail on OS X

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8639]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8639|
|Type|defect|
|Created|2016-07-20 14:37:39Z|
|Branch|https://github.com/twisted/twisted/tree/8639-cov-osx|

```
Traceback (most recent call last):

  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/twisted/test/test_iutils.py", line 265, in check

    self.assertEqual(out, dir)

  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 425, in assertEqual

    super(_Assertions, self).assertEqual(first, second, msg)

  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 515, in assertEqual

    assertion_func(first, second, msg=msg)

  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 508, in _baseAssertEqual

    raise self.failureException(msg)

twisted.trial.unittest.FailTest: '' != '/Users/travis/build/twisted/twisted/build/py27-coverage-posix/tmp/_trial_temp/twisted.test.test_iutils/ProcessUtilsTests/test_getProcessOutputAndValueDef/6PqRnn/temp'

twisted.test.test_iutils.ProcessUtilsTests.test_getProcessOutputAndValueDefaultPath

===============================================================================

[FAIL]

Traceback (most recent call last):

  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/twisted/test/test_iutils.py", line 252, in check

    self.assertEqual(result, 0)

  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 425, in assertEqual

    super(_Assertions, self).assertEqual(first, second, msg)

  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 515, in assertEqual

    assertion_func(first, second, msg=msg)

  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 508, in _baseAssertEqual

    raise self.failureException(msg)

twisted.trial.unittest.FailTest: 1 != 0

twisted.test.test_iutils.ProcessUtilsTests.test_getProcessValueDefaultPath

===============================================================================

[FAIL]

Traceback (most recent call last):

  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/twisted/test/test_process.py", line 1182, in check

    self.assertIn(b"Upon execvpe", errData)

  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 485, in assertIn

    % (containee, container))

twisted.trial.unittest.FailTest: 'Upon execvpe' not in ''

twisted.test.test_process.PosixProcessPTYTests.test_executionError

===============================================================================

[ERROR]

Traceback (most recent call last):

Failure: twisted.internet.utils._UnexpectedErrorOutput: got stderr: 'Traceback (most recent call last):\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site.py", line 703, in <module>\n    main()\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site.py", line 683, in main\n    paths_in_sys = addsitepackages(paths_in_sys)\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site.py", line 282, in addsitepackages\n    addsitedir(sitedir, known_paths)\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site.py", line 204, in addsitedir\n    addpackage(sitedir, name, known_paths)\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site.py", line 173, in addpackage\n    exec(line)\n  File "<string>", line 1, in <module>\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/coverage/__init__.py", line 13, in <module>\n    from coverage.control import Coverage, process_startup\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/coverage/control.py", line 14, in <module>\n    from coverage import env, files\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/coverage/files.py", line 158, in <module>\n    set_relative_directory()\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/coverage/files.py", line 27, in set_relative_directory\n    RELATIVE_DIR = os.path.normcase(abs_file(os.curdir) + os.sep)\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/site-packages/coverage/files.py", line 150, in abs_file\n    path = os.path.abspath(os.path.realpath(path))\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/posixpath.py", line 376, in realpath\n    return abspath(path)\n  File "/Users/travis/build/twisted/twisted/build/py27-coverage-posix/lib/python2.7/posixpath.py", line 364, in abspath\n    cwd = os.getcwd()\nOSError: [Errno 13] Permission denied\n'

twisted.test.test_iutils.ProcessUtilsTests.test_getProcessOutputDefaultPath
```

<details><summary>Searchable metadata</summary>

```
trac-id__8639 8639
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__8639_cov_osx 8639-cov-osx
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469025459739033 1469025459739033
changetime__1469199094375803 1469199094375803
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

PLease review. See the OS X 10.10 builder, one test fails on Travis for some reason, but that's Travis related, I believe.

https://buildbot.twistedmatrix.com/builders/osx10.10-py2.7-coverage/builds/295

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Looks good. Thanks!

It would help to leave a comment to explain why -S is required for this test... and why it was failing only on OSX

```
-S     : don't imply 'import site' on initialization
```

Please check my comment and merge!

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset f901015a804400179985a718a7667ee3100b47f2
```
#!CommitTicketReference repository="" revision="f901015a804400179985a718a7667ee3100b47f2"
Merge 8639-cov-osx: Fix tests under coverage on OS X

Author: hawkowl
Reviewer: adiroiban
Fixes: #8639
```
